cli/cmd/add_resource.go

```diff
@@ -77,29 +77,70 @@ keptn add-resource --project=rockshop --stage=production --service=shop --resour
 			},
 		}
 
-		endPoint.Path = path.Join(endPoint.Path, "configuration-service")
-
 		resourceHandler := apiutils.NewAuthenticatedResourceHandler(endPoint.String(), apiToken, "x-token", nil, endPoint.Scheme)
 
 		if endPointErr := checkEndPointStatus(endPoint.String()); endPointErr != nil {
 			return fmt.Errorf("Resource %s could not be uploaded: %s"+endPointErrorReasons,
 				*addResourceCmdParams.Resource, endPointErr)
 		}
 
-		if (addResourceCmdParams.Service != nil && *addResourceCmdParams.Service != "") && (addResourceCmdParams.Stage != nil && *addResourceCmdParams.Stage != "") {
+
+		// Prevent setting --stage and --all-stages at the same time
+		if (addResourceCmdParams.Stage != nil  && *addResourceCmdParams.Stage != "") &&
+			(addResourceCmdParams.AllStages != nil && *addResourceCmdParams.AllStages) {
+			return errors.New("Cannot use --stage and --all-stages at the same time")
+		}
+
+		// When setting --all-stages, project and service needs to be set
+		if (addResourceCmdParams.AllStages != nil && *addResourceCmdParams.AllStages) && (
+			(addResourceCmdParams.Service == nil || *addResourceCmdParams.Service == "") || (addResourceCmdParams.Project == nil || *addResourceCmdParams.Project == "")) {
+			return errors.New("--service and --project need to be supplied when using --all-stages")
+		}
+
+		// Handle different cases of adding resource to a projects default branch, stage branch, and/or service sub-directory
+		if (addResourceCmdParams.Service != nil && *addResourceCmdParams.Service != "") && (addResourceCmdParams.AllStages != nil && *addResourceCmdParams.AllStages) {
```

Comment:
provide helper functions which help checking the conditions... something like:
```
func assertFlagsSet(el... interface{}) bool {
	for e := range el {
		if !assertFlagSet(e) {
			return false
		}
	}
	return true
}


func assertFlagSet(s interface{}) bool {
	return &s != nil && s != ""
}
...
```

usage:

```
if (!assertFlagsSet(addResourceCmdParams.Stage,addResourceCmdParams.AllStages)) {
return errors.New("Cannot use --stage and --all-stages at the same time")
}
```